uefi-raw/src/protocol/shell.rs

```diff
@@ -10,30 +10,42 @@ use super::device_path::DevicePathProtocol;
 use super::file_system::FileInfo;
 use super::shell_params::ShellFileHandle;
 
+use bitflags::bitflags;
+
 /// List Entry for File Lists
 #[derive(Debug)]
 #[repr(C)]
-pub struct ListEntry<'a> {
-    pub f_link: *mut ListEntry<'a>,
-    pub b_link: *mut ListEntry<'a>,
+pub struct ListEntry {
+    pub f_link: *mut ListEntry,
+    pub b_link: *mut ListEntry,
 }
 
 /// ShellFileInfo for File Lists
 #[derive(Debug)]
 #[repr(C)]
-pub struct ShellFileInfo<'a> {
-    pub link: ListEntry<'a>,
+pub struct ShellFileInfo {
+    pub link: ListEntry,
     pub status: Status,
     pub full_name: *mut Char16,
     pub file_name: *mut Char16,
-    pub shell_file_handle: Handle,
-    pub file_info: FileInfo,
+    pub handle: ShellFileHandle,
+    pub info: FileInfo,
 }
 
 /// Used to specify where component names should be taken from
 pub type ShellDeviceNameFlags = u32;
-pub const DEVICE_NAME_USE_COMPONENT_NAME: u32 = 0x0000001;
-pub const DEVICE_NAME_USE_DEVICE_PATH: u32 = 0x0000002;
+
+bitflags! {
+    /// Specifies the source of the component name
+    #[repr(transparent)]
+    #[derive(Clone, Copy, Debug, Default, PartialEq, Eq, PartialOrd, Ord)]
+    pub struct DeviceName: u32 {
+        /// Use Component Name
+        const DEVICE_NAME_USE_COMPONENT_NAME = 0x0000001;
```

Comment:
Drop the `DEVICE_NAME_` prefix from these two constants, since it's already in the type name.